components/RecipeDetails.js

```diff
@@ -1,35 +1,53 @@
 import styled from "styled-components";
 import Image from "next/image";
 import Link from "next/link";
+import TrashBin from "public/svg/TrashBin.svg";
+import { useRouter } from "next/router";
+
+export default function RecipeDetails({ recipeDetails, onDeleteRecipe }) {
+  const router = useRouter();
+
+  function handleDeleteAndReroute() {
+    if (confirm("Are you sure you want to delete this recipe?")) {
+      onDeleteRecipe(recipeDetails.id);
+      router.push("/");
+    }
+  }
 
-export default function RecipeDetails({ recipeDetails }) {
   return (
-    <RecipeCard>
-      <h1> {recipeDetails.name}</h1>
-      <StyledImage
-        src={
-          recipeDetails.imageURL
-            ? recipeDetails.imageURL
-            : "/recipe-images/generic-food-3.png"
-        }
-        alt={recipeDetails.name}
-        width={300}
-        height={300}
-      />
-      <p>⏰ {recipeDetails.preparationTime}</p>
-      <ul>
-        {recipeDetails.ingredients.map(
-          (ingredient) =>
-            ingredient.ingredient && (
-              <li key={ingredient.ingredient}>
-                {ingredient.quantity} - {ingredient.ingredient}
-              </li>
-            )
-        )}
-      </ul>
-      <StyledDescription>{recipeDetails.description}</StyledDescription>
-      <StyledLink href="/">⬅ Back</StyledLink>
-    </RecipeCard>
+    <>
+      <RecipeCard>
+        <h1> {recipeDetails.name}</h1>
+        <StyledImage
+          src={
+            recipeDetails.imageURL
+              ? recipeDetails.imageURL
+              : "/recipe-images/generic-food-3.png"
+          }
+          alt={recipeDetails.name}
+          width={300}
+          height={300}
+        />
+        <p>⏰ {recipeDetails.preparationTime}</p>
+        <ul>
+          {recipeDetails.ingredients.map(
+            (ingredient) =>
+              ingredient.ingredient && (
+                <li key={ingredient.ingredient}>
+                  {ingredient.quantity} - {ingredient.ingredient}
+                </li>
+              )
+          )}
+        </ul>
+        <StyledDescription>{recipeDetails.description}</StyledDescription>
+      </RecipeCard>
+      <StyledActionElementsContainer>
+        <StyledLink href="/">⬅ Back</StyledLink>
+        <StyledTrashBin
+          onClick={() => handleDeleteAndReroute()}
+        ></StyledTrashBin>
```

Comment:
you should not make this icon directly clickable, instead nest it in a button. 

In general:
If its clickable it usually should be a button, but If it links to another route our website it should be an anchor.